yt_dlp/YoutubeDL.py

```diff
@@ -2163,6 +2166,18 @@ def can_merge():
             merger = FFmpegMergerPP(self)
             return merger.available and merger.can_merge()
 
+        def evaluate_formats(format_spec):
+            formats = info_dict.get('formats') or []
+            return list(self.build_format_selector(format_spec)({
+                'formats': formats,
+                'has_merged_format': any('none' not in (f.get('acodec'), f.get('vcodec')) for f in formats),
+                'incomplete_formats': (all(f.get('vcodec') == 'none' for f in formats)  # No formats with video
+                                       or all(f.get('acodec') == 'none' for f in formats)),  # OR, No formats with audio
+            }))
```

Comment:
Imo, this is complex enough for DRY and should be pulled out to a method:
```suggestion
        def _select_formats(self, formats, selector)
            return list(selector({
                'formats': formats,
                'has_merged_format': any('none' not in (f.get('acodec'), f.get('vcodec')) for f in formats),
                'incomplete_formats': (all(f.get('vcodec') == 'none' for f in formats)  # No formats with video
                                       or all(f.get('acodec') == 'none' for f in formats)),  # OR, No formats with audio
            }))
```